src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py

```diff
@@ -560,14 +561,24 @@ def flexible_server_update_custom_func(cmd, client, instance,
     if auto_grow:
         instance.storage.auto_grow = auto_grow
 
+    if public_access:
+        # set the public_access here only when public_access's lower class equals 'disabled' and 'enabled'
+        if str(public_access).lower() == 'Disabled'.lower():
+            instance.network.public_network_access = 'Disabled'
+        elif str(public_access).lower() == 'Enabled'.lower():
+            instance.network.public_network_access = 'Enabled'
+        else:
+            raise CLIError('Invalid public_access value for updating. Allowed values are Enabled or Disabled.')
```

Comment:
You'd better define another validator for your new public_access_arg_type, here's the original validator FYI:
https://github.com/Azure/azure-cli/blob/8b675e42ba3722114cc93df2e313b27ee2dfe044/src/azure-cli/azure/cli/command_modules/rdbms/validators.py#L442-L453

Reply:
remove this code and restrict input